app/middlewares/session.js

```diff
@@ -1,11 +1,23 @@
-const { validateToken } = require('../utils');
+const moment = require('moment');
+
+const { validateToken, decodedToken } = require('../utils');
 const errors = require('../errors');
+const { findUserBy } = require('../services/users');
 
 exports.tokenValidator = (req, res, next) => {
   const { token } = req.headers;
   if (token) {
     return validateToken(token)
-      .then(() => next())
+      .then(() => {
+        const { email, iat } = decodedToken(token);
+        return findUserBy({ email }, ['allowedDate']).then(({ allowedDate }) => {
+          const dateToken = moment.unix(iat);
```

Comment:
Is more easy ; moment(allowedDate), moment(iat) and moment.diff(moment1,moment2)